File: app/src/main/java/com/yuntongxun/mysurfaceviewdemo/MainActivity.java
package com.yuntongxun.mysurfaceviewdemo;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.view.WindowManager;
import android.widget.Button;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {


    private BulletChatView bulletChatView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        getWindow().addFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
        initView();
    }

    private void initView() {
        Button startBulletChat = (Button) findViewById(R.id.btn_star_bulletchat);
        Button stopBulletChat = (Button) findViewById(R.id.btn_stop_bulletchat);
        Button clearBulletChat = (Button) findViewById(R.id.btn_clear_bulletchat);
        Button addBulletChat = (Button) findViewById(R.id.btn_add_bulletchat);
        bulletChatView = (BulletChatView) findViewById(R.id.bcv_view);
        startBulletChat.setOnClickListener(this);
        stopBulletChat.setOnClickListener(this);
        clearBulletChat.setOnClickListener(this);
        addBulletChat.setOnClickListener(this);
    }


    @Override
    public void onClick(View v) {
        int id = v.getId();
        if (id == R.id.btn_star_bulletchat) {
            bulletChatView.startBulletChat();
        } else if (id == R.id.btn_stop_bulletchat) {
            bulletChatView.stopBulletChat();
        } else if (id == R.id.btn_clear_bulletchat) {
            bulletChatView.clearView();
        } else if (id == R.id.btn_add_bulletchat) {
            BulletChatContentInfo bulletChatContentInfo = new BulletChatContentInfo();
            bulletChatContentInfo.setBulletCharMessage("飞机来了。。。。。。。。。。。。。。");
            bulletChatView.insertBulletChatData(bulletChatContentInfo);
        }
    }
}

File: study_view/src/main/java/com/view/cornerImage/MainActivity.kt
package com.view.cornerImage

import android.animation.AnimatorSet
import android.animation.ObjectAnimator
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView
import com.sn.study_pic.R

class MainActivity : AppCompatActivity() {


    var mTextView: TextView? = null;

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_image)
//        mTextView = findViewById<TextView>(R.id.textView);
//        mTextView!!.setOnClickListener { startAnimi() }
//        startAnimi();
//        CornerImage(baseContext);
    }

    fun startAnimi() {
        val translationX: Float = mTextView?.translationX?.toFloat() ?: 0f
        val toFloat = mTextView?.width?.toFloat() ?: 0f
        val scaleX = ObjectAnimator.ofFloat(mTextView, "scaleX", 1f, 2f, 1f);
        val scaleY = ObjectAnimator.ofFloat(mTextView, "scaleY", 1f, 2f, 1f);
        val animatorSet = AnimatorSet();
        animatorSet.play(scaleX).with(scaleY);
        animatorSet.duration = 2000;
        animatorSet.start();
    }
}